src/include/interval_set.h

```diff
@@ -46,6 +46,14 @@ class interval_set {
   class iterator : public std::iterator <std::forward_iterator_tag, T>
   {
     public:
+        using Map = C<T, T>;
+        using value_type = typename Map::value_type;
+        using offset_type = T;
+        using length_type = T;
+        using reference = value_type&;
+        using const_reference = const value_type&;
+        using size_type = typename Map::size_type;
```

Comment:
neither is this needed.

Reply:
@tchaikov 
I have added a new commit, which minimizes the PR and adds the suggestion made by @dillaman 
removing the `std::iterator<....>` leads to complains about missing the `reference` definition in several incantiations I tried.
So I think this is what we should commit to fix the difference between libc++ and libstdc++
If you agree I'll squeeze the 2 commits into 1

Reply:
```diff
diff --git a/src/include/interval_set.h b/src/include/interval_set.h
index 5ff065cd55b..56e1222a4bc 100644
--- a/src/include/interval_set.h
+++ b/src/include/interval_set.h
@@ -43,9 +43,15 @@ class interval_set {

   class const_iterator;

-  class iterator : public std::iterator <std::forward_iterator_tag, T>
+  class iterator
   {
-    public:
+      public:
+        using difference_type = ssize_t;
+        using value_type = typename Map::value_type;
+        using pointer = typename Map::value_type*;
+        using reference = typename Map::value_type&;
+        using iterator_category = std::forward_iterator_tag;
+
         explicit iterator(typename Map::iterator iter)
           : _iter(iter)
         { }
@@ -107,9 +113,15 @@ class interval_set {
     friend class interval_set;
   };

-  class const_iterator : public std::iterator <std::forward_iterator_tag, T>
+  class const_iterator
   {
     public:
+        using difference_type = ssize_t;
+        using value_type = const typename Map::value_type;
+        using pointer = const typename Map::value_type*;
+        using reference = const typename Map::value_type&;
+        using iterator_category = std::forward_iterator_tag;
+
         explicit const_iterator(typename Map::const_iterator iter)
           : _iter(iter)
         { }
@@ -130,7 +142,7 @@ class interval_set {
         }

         // Dereference this iterator to get a pair.
-        const_reference operator*() const {
+        reference operator*() const {
           return *_iter;
         }
```

@wjwithagen i created the patch above i think it is what i suggested.

Reply:
> @wjwithagen i created the patch above. i think it is what i suggest.

I tried something almost similar like this, which did not work.
But this issue has proven very subtile, so it very well could please clang.
I'll run it thru the compiler.

Reply:
@tchaikov 
I've run you suggestions thru the builder, but the change for `reference operator*` gives arrors.
Leaving it out, works... So I've submitted the change without that piece of code.

Reply:
okay, i will submit another PR to fix `const_iterator`.